tests/test_client_legacy.py

```diff
@@ -245,45 +247,35 @@ def test_streaming_send_message(mock_e2b_api_key_none, client: RESTClient, agent
     inner_thoughts_count = 0
     # 2. Check that the agent runs `send_message`
     send_message_ran = False
-    # 3. Check that we get all the start/stop/end tokens we want
-    #    This includes all of the MessageStreamStatus enums
-    done_gen = False
-    done_step = False
+    # 3. Check that we get the end token we want (StreamDoneStatus)
     done = False
 
     # print(response)
     assert response, "Sending message failed"
     for chunk in response:
-        assert isinstance(chunk, LettaStreamingResponse)
-        if isinstance(chunk, ReasoningMessage) and chunk.reasoning and chunk.reasoning != "":
-            inner_thoughts_exist = True
-            inner_thoughts_count += 1
-        if isinstance(chunk, ToolCallMessage) and chunk.tool_call and chunk.tool_call.name == "send_message":
-            send_message_ran = True
-        if isinstance(chunk, MessageStreamStatus):
-            if chunk == MessageStreamStatus.done:
-                assert not done, "Message stream already done"
-                done = True
-            elif chunk == MessageStreamStatus.done_step:
-                assert not done_step, "Message stream already done step"
-                done_step = True
-            elif chunk == MessageStreamStatus.done_generation:
-                assert not done_gen, "Message stream already done generation"
-                done_gen = True
-        if isinstance(chunk, LettaUsageStatistics):
-            # Some rough metrics for a reasonable usage pattern
-            assert chunk.step_count == 1
-            assert chunk.completion_tokens > 10
-            assert chunk.prompt_tokens > 1000
-            assert chunk.total_tokens > 1000
+        if isinstance(chunk, LettaMessage):
+            if isinstance(chunk, ReasoningMessage) and chunk.reasoning and chunk.reasoning != "":
+                inner_thoughts_exist = True
+                inner_thoughts_count += 1
+            if isinstance(chunk, ToolCallMessage) and chunk.tool_call and chunk.tool_call.name == "send_message":
+                send_message_ran = True
+            if isinstance(chunk, UsageMessage):
+                # Some rough metrics for a reasonable usage pattern
+                assert chunk.usage.step_count == 1
+                assert chunk.usage.completion_tokens > 10
+                assert chunk.usage.prompt_tokens > 1000
+                assert chunk.usage.total_tokens > 1000
+        elif chunk == OPENAI_SSE_DONE:
+            assert not done, "Message stream already done"
+            done = True
+        else:
+            assert isinstance(chunk, LettaStreamingResponse)
```

Comment:
@carenthomas why did you move the `assert` into an `else`? Can we keep it before the `if isinstance(chunk, LettaMessage):`?